solver/src/liquidity.rs

```diff
@@ -21,6 +18,23 @@ pub enum Liquidity {
     Limit(LimitOrder),
     Amm(AmmOrder),
 }
+
+/// A trait associating some liquidity model to how it is executed. This allows
+/// different liquidity types to be modeled the same way.
+pub trait LiquidityModel {
+    type Execution;
+
+    fn settlement_handling(&self) -> &dyn SettlementHandling<Self>;
+}
+
+/// Specifies how a liquidity exectution gets encoded into a settlement.
+pub trait SettlementHandling<Model>: Send + Sync
+where
+    Model: LiquidityModel,
+{
+    fn encode(&self, execution: Model::Execution, encoder: &mut SettlementEncoder) -> Result<()>;
```

Comment:
I really like this extra layer of indirection. While it might be a little harder to understand, it really decouples liquidity from solvers and as such is a great step towards allowing independent liquidity/solver development.

Reply:
The other thing I like about it, is we can now add a `unwrap_native_token` to the `SettlementEncoder` and only modify the `impl SettlementHandling<LimitOrder> for Order` implementation (since we don't need to extend the trait, but only the implementation of where it applies). In theory, if there were other liquidity that were modeled as `LimitOrder`s, they would also not need to be changed :smile:.